### PR TITLE
[Feature] Project Ideas

### LINKED ISSUE #581: [Feature] Add Projects Section
### Description

Brief -
Currently website lack the practical implementations/knowledge resource section.
I want to add PROJECTS section on the Resources section of website, It will have Project Tasks of the respective languages/resources.

Solution -
By addition of projects section, development beginners can get a path to begin with after learning from the resources and gain hand on experience For example, it will have challenging beginner projects ideas that a beginner can begin with like basic portfolio website, different CSS styling projects etc.
Basically by projects section, a learner can get a path to implement his knowledge that he/she has gained. As making projects is equally important while learning.

Additional context
Many beginners don't know a path and what projects to be done after learning, this feature will be a great addition to the website and its resource. Solution/Links for the codes of projects will also be there, so if a learner get stuck, he can get a idea/solutions.

### Checklist

- [X] I have checked the existing issues
- [X] I have read the [Contributing Guidelines](https://github.com/rupali-codes/LinksHub/blob/main/CONTRIBUTING.md)
- [X] I am willing to work on this issue (optional)
- [X] I am a GSSoC'23 contributor

### PR DESCRIPTION
**Pull request by aditya9277**
## Fixes Issue

Closes #581 

## Changes proposed

[Feature] - Added Projects Subcategory under Resources Category.

[Add] - Added Various resources under Projects Subcategory to initialize it.

## Screenshots

pfeature

presources

## Note to reviewers

As discussed, I added a Projects subcategory under Resources category and Initialized it too by adding various resources.
Thanks and would be glad by your response.

### CODE REVIEW

**Review by rupali-codes — CHANGES_REQUESTED** [review]
Thanks. Could  you name it **Project Ideas** instead?

File: database/resources/project-resources.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+{
+    "name": "React.Js Projects",
+    "description": "This website provides free eight React.Js projects with source code.",
+    "url": "https://www.freecodecamp.org/news/react-projects-to-improve-your-skills/",
+    "category": "resources",
+    "subcategory": "project-resources"
+},
+{
+    "name": "Machine Learning Projects",
+    "description": "This website provides free Machine Learning Projects from beginners to advanced with source code.",
```
Comment (rupali-codes) [inline database/resources/project-resources.json:31]:
```suggestion
    "description": "This website provides free Machine Learning Projects from beginner to advanced level with source code.",
```

File: database/resources/project-resources.json [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    "url": "https://www.interviewbit.com/blog/machine-learning-projects/",
+    "category": "resources",
+    "subcategory": "project-resources"
+},
+{
+    "name": "Python Projects",
+    "description": "This website provides free twenty five beginner friendly python projects in Python with source code.",
```
Comment (rupali-codes) [inline database/resources/project-resources.json:38]:
```suggestion
    "description": "This website provides free twenty five beginner friendly Python projects  with source code.",
```

File: database/resources/project-resources.json
Comment (rupali-codes) [inline database/resources/project-resources.json:1]: Also, the code doesn't look like formatted, could you format the code please?

Comment (aditya9277) [reply database/resources/project-resources.json:1]: Hey, i have applied the desired formatting also. Please check @rupali-codes

**Comment by rupali-codes** [comment]
@aditya9277 not just the PR, rename it like from Projects to Project Ideas :)

File: database/resources/project-ideas.json [hunk 3]
```diff
@@ -0,0 +1,43 @@
+[{    
+    "name": "Flutter Projects",
+    "description": "This website provides free fifteen flutter based projects with source code.",
```
Comment (CBID2) [inline database/resources/project-ideas.json:3]: [Typo]
```suggestion
    "description": "This website provides free fifteen flutter-based projects with source code.",
```

File: database/resources/project-ideas.json [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  {
+    "name": "Machine Learning Projects",
+    "description": "This website provides free Machine Learning Projects from beginners to advanced with source code.",
+    "url": "https://www.interviewbit.com/blog/machine-learning-projects/",
+    "category": "resources",
+    "subcategory": "project-ideas"
+  },
+  {
+    "name": "Python Projects",
+    "description": "This website provides free twenty five beginner friendly python projects in Python with source code.",
```
Comment (CBID2) [inline database/resources/project-ideas.json:38]: [Typo]
```suggestion
    "description": "This website provides free twenty-five beginner-friendly Python projects with source code.",
```

File: database/resources/project-ideas.json [hunk 5]
```diff
@@ -0,0 +1,43 @@
+[{    
+    "name": "Flutter Projects",
+    "description": "This website provides free fifteen flutter based projects with source code.",
+    "url": "https://www.interviewbit.com/blog/flutter-projects/",
+    "category": "resources",
+    "subcategory": "project-ideas"
+  },
+  {
+    "name": "HTML & CSS Projects",
+    "description": "This website provides free ten beginner friendly projects in HTML & CSS with source code.",
```
Comment (CBID2) [inline database/resources/project-ideas.json:10]: [Typo]
```suggestion
    "description": "This website provides free ten beginner-friendly projects in HTML & CSS with source code.",
```

File: database/resources/project-ideas.json [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  {
+    "name": "React.Js Projects",
+    "description": "This website provides free eight React.Js projects with source code.",
+    "url": "https://www.freecodecamp.org/news/react-projects-to-improve-your-skills/",
+    "category": "resources",
+    "subcategory": "project-ideas"
+  },
+  {
+    "name": "Machine Learning Projects",
+    "description": "This website provides free Machine Learning Projects from beginners to advanced with source code.",
```
Comment (CBID2) [inline database/resources/project-ideas.json:31]: [Typo]
```suggestion
    "description": "This website provides free Machine Learning projects for beginners to advanced with source code.",
```

**Review by aditya9277 — COMMENTED** [review]
Changes done for Projects to Project Ideas

**Review by CBID2 — CHANGES_REQUESTED** [review]
Hi @aditya927! :) Your PR is off to a great start! I found some typos in the description, so fix those and I'll review your PR again. In the future, consider running through the descriptions of your chosen sources through spell check or Grammarly before putting them in your PRs. Keep it up! :)

**Comment by aditya9277** [comment]
Thanks @CBID2, from future i would definitely keep these points in mind and will run a check too
Now I have fixed the typos in the description, please review my PR again.

**Review by CBID2 — APPROVED** [review]
Well done @aditya9277! LGTM! :)

**Comment by aditya9277** [comment]
Thanks a lot @CBID2, it was my first contribution to the project and learnt many things.
So now, is there something more to do in this PR or its done?

**Review by rupali-codes — APPROVED** [review]
looks great! thanks a lot :)

**Comment by CBID2** [comment]
@rupali-codes, we're going to need your force push

**Pull request merged**